### DISCUSSION TITLE
API key for self-hosted instance?

### DISCUSSION THREAD

**Discussion by arladmin** [plausible/analytics#747] [Self-Hosted Support]
This [documentation](https://plausible.io/docs/stats-api) provides the method for retrieving the API key only for accounts with plausible.io. 

How to get the API key for self-hosted accounts?

**Comment by metmarkosaric** [comment]
thanks @arladmin! It's not possible at this time for self-hosted. 

Self-hosted version is not updated as frequently as the hosted version which changes daily. API is a brand new feature we're testing right now. Think of the self-hosted as a long term release to help us and help you not need to upgrade things all the time and risk breaking stuff. We stack all the new features and fixes that have been battle tested on the hosted version and release a new self-hosted version twice a year. you can learn more here: https://plausible.io/blog/building-open-source

**Reply by arladmin** [reply]
Hi @metmarkosaric , thanks for your message and for sharing the above info.

Wasn't aware of Plausible's only bi-yearly update policy. 
That's a bit of a shock, i must say, as waiting for 6 months for the next improvement/enhancement update is a bit unfamiliar concept in the context of today's fast-paced software (including open-source) industry.

Anyway, i guess the API feature would then be released for the self-hosted version not before 5 months from now, right?

**Reply by metmarkosaric** [reply]
you're welcome. 6 months is a pretty standard long term support release schedule for more complex open source projects.

about when it will be released to self-hosted release depends on our beta testing in the cloud and when we feel the feature is mature and ready. stay tuned.

**Reply by arladmin** [reply]
Strange. Would expect it to be the other way around, as usually is (release schedule based on feature maturity: open source vs proprietary/hosted).

Is there any recommended workaround (even unsafe) to use the APIs.
Primary use-case is to get the stats for the day programmatically, from an external system.

**Reply by ukutaht** [reply]
Feel free to build your own docker image from source. The `master` branch includes the beta version of the API.

**Reply by metmarkosaric** [reply]
> Strange. Would expect it to be the other way around, as usually is (release schedule based on feature maturity: open source vs proprietary/hosted).

Just a note that Plausible is not proprietary software neither in the cloud nor when self-hosted

**Reply by arladmin** [reply]
> Feel free to build your own docker image from source. The `master` branch includes the beta version of the API.

Thanks for the tip! Will try that.

**Reply by bhavin192** [reply]
Seems like there is a `master` and `dev` container image tag as well. I think one can use those if they want to use / test the latest code?

**Reply by ukutaht** [reply]
Yes the `master` tag is updated via CI on push.

It is intended for internal use so no guarantees that the tag will be updated or kept around in the future.

**Reply by omar7c1** [reply]
It really seems like they have paid software and self hosting, not that that's bad or a problem. but it would be good to give the option to install the stable version in docker and the test version to make it more accessible that new programmers enter your project

I am trying to enter the world of elixir programming since its project sounds interesting and in my field but the option that you install in docker is an inferior version to other tools that I currently use, but the paid project sounds quite interesting

**Reply by metmarkosaric** [reply]
not sure what you mean @omar7c1. the self-hosted and hosted are the same project with same code and same principles. both fully free and open source software.

we've just made self-hosted more convenient so you don't need to waste time needing to update it every day on your own server as it's a big task with risks involved. this way you just update it twice a year so you can focus your time on important things such as running your website rather than on running your analytics. 

everything is clearly explained on the landing page and in the linked post that has details on the frequency and we've got great feedback since making this change and several larger companies have started using self-hosted since: https://plausible.io/self-hosted-web-analytics

and if you want to hack and play around? feel free to build your own docker image from source. that's possible too

**Reply by omar7c1** [reply]
I think that the tools they have stable on their paid server are very interesting, the part in the self-hosted manual is not that interesting .. what I did to test it is change: docker-compose.yml
   plausible:
     image: plausible / analytics: master
(delete) image: plausible / analytics: latest

What is strange is that the self hosted is the stable one and the paid version is beta: S. usually the opposite

The project is very interesting to me and I am trying to understand this programming language more, so I installed it more than anything to learn how to do things in a larger project

this is not an attack quite the opposite! i love your project

**Reply by metmarkosaric** [reply]
thanks! have fun exploring!
